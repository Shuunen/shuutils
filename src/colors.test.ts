import { expect, it } from 'bun:test'
import {
  addColorCode,
  bgBlack,
  bgBlue,
  bgCyan,
  bgGray,
  bgGreen,
  bgMagenta,
  bgRed,
  bgWhite,
  bgYellow,
  black,
  blue,
  bold,
  cyan,
  dim,
  gray,
  green,
  inverse,
  isFirefox,
  italic,
  magenta,
  overline,
  red,
  reset,
  strikeThrough,
  underline,
  white,
  yellow,
} from './shuutils'

it('is not Firefox', () => {
  expect(isFirefox()).toBe(false)
})

it('add color code', () => {
  expect(addColorCode(22, 32, 'test')).toMatchSnapshot()
})

it('color reset', () => {
  expect(reset('test')).toMatchSnapshot()
})
it('color bold', () => {
  expect(bold('test')).toMatchSnapshot()
})
it('color dim', () => {
  expect(dim('test')).toMatchSnapshot()
})
it('color italic', () => {
  expect(italic('test')).toMatchSnapshot()
})
it('color underline', () => {
  expect(underline('test')).toMatchSnapshot()
})
it('color overline', () => {
  expect(overline('test')).toMatchSnapshot()
})
it('color inverse', () => {
  expect(inverse('test')).toMatchSnapshot()
})
it('color strikeThrough', () => {
  expect(strikeThrough('test')).toMatchSnapshot()
})
it('color black', () => {
  expect(black('test')).toMatchSnapshot()
})
it('color red', () => {
  expect(red('test')).toMatchSnapshot()
})
it('color green', () => {
  expect(green('test')).toMatchSnapshot()
})
it('color yellow', () => {
  expect(yellow('test')).toMatchSnapshot()
})
it('color blue', () => {
  expect(blue('test')).toMatchSnapshot()
})
it('color magenta', () => {
  expect(magenta('test')).toMatchSnapshot()
})
it('color cyan', () => {
  expect(cyan('test')).toMatchSnapshot()
})
it('color white', () => {
  expect(white('test')).toMatchSnapshot()
})
it('color gray', () => {
  expect(gray('test')).toMatchSnapshot()
})
it('color bgBlack', () => {
  expect(bgBlack('test')).toMatchSnapshot()
})
it('color bgRed', () => {
  expect(bgRed('test')).toMatchSnapshot()
})
it('color bgGreen', () => {
  expect(bgGreen('test')).toMatchSnapshot()
})
it('color bgYellow', () => {
  expect(bgYellow('test')).toMatchSnapshot()
})
it('color bgBlue', () => {
  expect(bgBlue('test')).toMatchSnapshot()
})
it('color bgMagenta', () => {
  expect(bgMagenta('test')).toMatchSnapshot()
})
it('color bgCyan', () => {
  expect(bgCyan('test')).toMatchSnapshot()
})
it('color bgWhite', () => {
  expect(bgWhite('test')).toMatchSnapshot()
})
it('color bgGray', () => {
  expect(bgGray('test')).toMatchSnapshot()
})
