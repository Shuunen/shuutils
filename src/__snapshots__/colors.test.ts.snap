// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`add color code 1`] = `"\x1B[22mtest\x1B[32m"`;

exports[`color reset 1`] = `"\x1B[0mtest\x1B[0m"`;

exports[`color bold 1`] = `"\x1B[1mtest\x1B[22m"`;

exports[`color dim 1`] = `"\x1B[2mtest\x1B[22m"`;

exports[`color italic 1`] = `"\x1B[3mtest\x1B[23m"`;

exports[`color underline 1`] = `"\x1B[4mtest\x1B[24m"`;

exports[`color overline 1`] = `"\x1B[53mtest\x1B[55m"`;

exports[`color inverse 1`] = `"\x1B[7mtest\x1B[27m"`;

exports[`color strikeThrough 1`] = `"\x1B[9mtest\x1B[29m"`;

exports[`color black 1`] = `"\x1B[30mtest\x1B[39m"`;

exports[`color red 1`] = `"\x1B[31mtest\x1B[39m"`;

exports[`color green 1`] = `"\x1B[32mtest\x1B[39m"`;

exports[`color yellow 1`] = `"\x1B[33mtest\x1B[39m"`;

exports[`color blue 1`] = `"\x1B[34mtest\x1B[39m"`;

exports[`color magenta 1`] = `"\x1B[35mtest\x1B[39m"`;

exports[`color cyan 1`] = `"\x1B[36mtest\x1B[39m"`;

exports[`color white 1`] = `"\x1B[37mtest\x1B[39m"`;

exports[`color gray 1`] = `"\x1B[90mtest\x1B[39m"`;

exports[`color bgBlack 1`] = `"\x1B[40mtest\x1B[49m"`;

exports[`color bgRed 1`] = `"\x1B[41mtest\x1B[49m"`;

exports[`color bgGreen 1`] = `"\x1B[42mtest\x1B[49m"`;

exports[`color bgYellow 1`] = `"\x1B[43mtest\x1B[49m"`;

exports[`color bgBlue 1`] = `"\x1B[44mtest\x1B[49m"`;

exports[`color bgMagenta 1`] = `"\x1B[45mtest\x1B[49m"`;

exports[`color bgCyan 1`] = `"\x1B[46mtest\x1B[49m"`;

exports[`color bgWhite 1`] = `"\x1B[47mtest\x1B[49m"`;

exports[`color bgGray 1`] = `"\x1B[100mtest\x1B[49m"`;
